frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -540,30 +544,49 @@ private void serveAttachmentFileLicenses(ResourceRequest request, ResourceRespon
             // In addition we remember the license information for exclusion later on
             Map<String, LicenseNameWithText> licenseStore = Maps.newHashMap();
             List<Map<String, String>> licenses = Lists.newArrayList();
-            licenseInfos.forEach(licenseInfo -> {
-                List<Map<String, String>> licenesAsObject = licenseInfo.licenseInfo.licenseNamesWithTexts.stream()
-                        .filter(licenseNameWithText -> {
-                            return !Strings.isNullOrEmpty(licenseNameWithText.getLicenseName())
-                                    || !Strings.isNullOrEmpty(licenseNameWithText.getLicenseText());
-                        }).map(licenseNameWithText -> {
-                            // Since the license has no good identifier, we create one and store the license
-                            // in the session. If the final report is generated, we use the identifier to
-                            // identify the licenses to be excluded
-                            // FIXME: this could be changed if we scan the attachments once after uploading
-                            // and store them as own entity
-                            String key = UUID.randomUUID().toString();
-                            licenseStore.put(key, licenseNameWithText);
-
-                            Map<String, String> data = Maps.newHashMap();
-                            data.put(LICENSE_NAME_WITH_TEXT_KEY, key);
-                            data.put(LICENSE_NAME_WITH_TEXT_NAME, Strings.isNullOrEmpty(licenseNameWithText.getLicenseName()) ? EMPTY
-                                    : licenseNameWithText.getLicenseName());
-                            data.put(LICENSE_NAME_WITH_TEXT_TEXT, licenseNameWithText.getLicenseText());
-                            return data;
-                        }).collect(Collectors.toList());
-
-                licenses.addAll(licenesAsObject);
-            });
+            licenseInfos.forEach(licenseInfoResult -> {
+                        switch (licenseInfoResult.getStatus()){
```

Comment:
I think it would be good to move that lambda to its own named function.

Reply:
good point